cardano-node/Cardano/Node/CLI.hs

```diff
@@ -93,28 +110,43 @@ data SomeProtocol where
   SomeProtocol :: (RunDemo blk, TraceConstraints blk)
                => Consensus.Protocol blk -> SomeProtocol
 
-fromProtocol :: Protocol -> IO SomeProtocol
-fromProtocol BFT =
+fromProtocol :: CardanoConfiguration -> Protocol -> IO SomeProtocol
+fromProtocol _ BFT =
     case Consensus.runProtocol p of
       Dict -> return $ SomeProtocol p
   where
     p = ProtocolMockBFT defaultSecurityParam
-fromProtocol Praos =
+fromProtocol _ Praos =
     case Consensus.runProtocol p of
       Dict -> return $ SomeProtocol p
   where
     p = ProtocolMockPraos defaultDemoPraosParams
-fromProtocol MockPBFT =
+fromProtocol _ MockPBFT =
     case Consensus.runProtocol p of
       Dict -> return $ SomeProtocol p
   where
     p = ProtocolMockPBFT defaultDemoPBftParams
-fromProtocol RealPBFT =
+fromProtocol cc@CardanoConfiguration{..} RealPBFT = do
+    let Genesis{..} = coGenesis ccCore
+        genHash = either (throw . ConfigurationError) id $ -- XXX: need a proper error tag
```

Comment:
Is this left for later, or did you intended to do this in this PR?

Reply:
Since it'd involve changing `cardano-shell`, I think `ConfigurationError` might be sufficient for now.  I removed the TODO.

Reply:
It's still throwing in a pure code. If you'd like to be 100% kosher, then using `throwIO` is a nicer alternative.